enumerable_module.rb

```diff
@@ -0,0 +1,316 @@
+    extend Snippet
+    if block_given?
+      return my_select { |item, value| yield(item, value) }.length.zero? if is_a?(Hash)
+
+      my_select { |item| yield(item) }.length.zero?
+    elsif !arg.nil?
+      none_snippet(arg)
+    else
+      return my_select { |item, _value| item }.length.zero? if is_a?(Hash)
+
+      my_select { |item| item }.length.zero?
+    end
+  end
+
+  def my_count(arg = nil)
+    arr = []
+    if block_given?
+      my_each { |i| arr << i if yield i }
+    elsif !arg.nil?
+      my_select { |i| arr << i if i == arg }
+    else
+      arr = self
+    end
+    arr.size
+  end
+
+  def my_map(proc = nil)
+    return enum_for :my_map unless block_given?
+
+    arr = []
+    proc ? my_each { |value| arr << proc.call(value) } : my_each { |value| arr << yield(value) }
+    arr
+  end
+
+  def my_inject(*args)
+    arr = is_a?(Range) ? to_a : self
+    acc = args[0]
+    if block_given?
+      n = args[0] ? 0 : 1
+      acc ||= arr[0]
+      n.upto(arr.length - 1) { |value| acc = yield(acc, arr[value]) }
+    elsif args[0].is_a?(Symbol) || args[0].is_a?(Proc)
+      acc = arr[0]
+      1.upto(arr.length - 1) { |value| acc = args[0].to_proc.call(acc, arr[value]) }
+    else
+      arr.insert(0, args[0])
+      1.upto(arr.length - 1) { |value| acc = args[1].to_proc.call(acc, arr[value]) }
+    end
+    acc
+  end
+
+  def multiply_els
+    my_inject(100) { |product, n| product * n }
+  end
+end
+
+module Snippet
+  def all_snippet(arg)
+    case arg
+    when Class
+      my_select { |item| item.is_a?(arg) }.length == length
+    when Regexp
+      my_select { |item| item =~ arg }.length == length
+    else
+      my_select { |item| item.equal?(arg) }.length == length
+    end
+  end
+
+  def any_snippet(arg)
+    case arg
+    when Class
+      my_select { |item| item.is_a?(arg) }.length.positive?
+    when Regexp
+      my_select { |item| item =~ arg }.length.positive?
+    else
+      my_select { |item| item.equal?(arg) }.length.positive?
+    end
+  end
+
+  def none_snippet(arg)
+    case arg
+    when Class
+      my_select { |item| item.is_a?(arg) }.length.zero?
+    when Regexp
+      my_select { |item| item =~ arg }.length.zero?
+    else
+      my_select { |item| item.equal?(arg) }.length.zero?
+    end
+  end
+end
+
+# Examples to test methods with
+
+# arr = %w[a b c c d]
+# words = %w[dog door rod blade]
+# words[0] = 5
+# arr1 = [nil, 'a', 'b']
+# arrnum = [2, 2, 3, 4, 5, 6]
+# hash = { a: 1, b: 2, c: 3, d: 4 }
+# false_array = [1, false, 'hi', []]
+# false_array1 = [nil, false, nil, false]
+# false_array2 = [true, false, true, false]
+# array = [
+#   1, 4, 6, 5, 1, 1, 3, 0, 7, 4, 0, 6, 5, 3, 1, 1, 0,
+#   7, 4, 2, 3, 7, 1, 3, 1, 8, 1, 6, 6, 0, 7, 2, 8, 5,
+#   5, 0, 1, 5, 5, 1, 0, 1, 8, 2, 2, 3, 5, 4, 7, 6, 5,
+#   1, 4, 8, 1, 7, 6, 7, 8, 0, 3, 6, 8, 7, 1, 3, 6, 4,
+#   6, 6, 3, 1, 6, 8, 3, 3, 7, 7, 7, 5, 7, 4, 0, 0, 6,
+#   5, 6, 8, 5, 6, 5, 6, 8, 0, 4, 1, 1, 5, 4, 3
+# ]
+# range = Range.new(1, 50)
```

Comment:
Please remove commented out code 🙏 as it’s better to delete commented code. If you need to review a previous version of your code, use git history 🙂